networkx/algorithms/shortest_paths/weighted.py

```diff
@@ -174,7 +174,7 @@ def single_source_dijkstra_path(G, source, cutoff=None, weight='weight'):
     """
     (length, path) = single_source_dijkstra(
         G, source, cutoff=cutoff, weight=weight)
-    return path
+    return iter(path.items())
```

Comment:
Why `iter` necessary here? Isn't `path.items()` already iterable? That should be enough for the user to write code that uses the return value of this function, expecting an iterable.